File: tools/benchmark/oracle-fixtures/sample.go
package ledger

import "fmt"

const MaxRetries = 3

var DefaultName = "conducks"

type Account struct {
	ID      uint64
	Balance int64
}

type Ledger interface {
	Post(amount int64) error
}

type AccountID = uint64

func NewAccount(id uint64) *Account {
	return &Account{ID: id}
}

func (a *Account) Post(amount int64) error {
	a.Balance += amount
	return nil
}

func Transfer(from *Account, to *Account, amount int64) bool {
	if amount <= 0 {
		return false
	}
	fmt.Println("transfer")
	return true
}

File: tools/benchmark/oracle-fixtures/sample.c
#include <stdio.h>

#define MAX_RETRIES 3

typedef unsigned long AccountId;

struct Account {
    AccountId id;
    long balance;
};

enum Status { ACTIVE, FROZEN };

union Raw {
    unsigned int int_value;
    float float_value;
};

static long default_balance = 0;

void post(struct Account *account, long amount) {
    account->balance += amount;
}

int transfer(struct Account *from, struct Account *to, long amount) {
    if (amount <= 0) return 0;
    post(from, -amount);
    post(to, amount);
    return 1;
}
